### PR TITLE
fix(jest-preset-mc-app): to use default v16 enzyme adapter

### PR DESCRIPTION
**Pull request by tdeekens**
#### Summary

This pull request suggests to default again to v16 of the enzyme adapter while offering a config option to opt into a newer version.

We noticed that the v17 adapter can't be as easily used as expeceted.

### CODE REVIEW

File: packages/jest-preset-mc-app/load-config.js [hunk 1]
```diff
@@ -3,6 +3,12 @@ const { cosmiconfigSync } = require('cosmiconfig');
 const moduleName = 'jest-preset-mc-app';
 const explorer = cosmiconfigSync(moduleName);
 
+function createListMergerWithDefaults(key) {
```
Comment (tdeekens) [inline packages/jest-preset-mc-app/load-config.js:6]: Unrelated: hoisting fanciness is not needed I think.

File: packages/jest-preset-mc-app/setup-test-framework.js [hunk 2]
```diff
@@ -13,7 +13,12 @@ const loadConfig = require('./load-config');
 
 const jestConfig = loadConfig();
 
-Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: true });
+Enzyme.configure({
+  adapter: jestConfig.EnzymeAdapter
```
Comment (tdeekens) [inline packages/jest-preset-mc-app/setup-test-framework.js:17]: So we default again the the old version and offer an option to pass in any adapter you want. An alternative is a config string e.g. v17 and v16 but then both dependencies need to reside here and it gives the impression we support any version.

**Review by adnasa — APPROVED** [review]
thanks for the follow up!

**Comment by tdeekens** [comment]
As another step the `EnzymeAdapter` should be required if it's desired and totally opt-in. This would be a breaking change though.

**Pull request merged**